.github/workflows/droid.yml

```diff
@@ -0,0 +1,69 @@
+name: Build for Android
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      release_type:
+        required: true
+        type: string
+    secrets:
+      GCP_CREDENTIALS:
+        required: true
+
+env:
+  NDK_ZIP: android-ndk-r26-linux.zip
+  TARGET: bin/viam-server-${{ inputs.release_type }}-android-aarch64
+
+jobs:
+  droid-build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
+
+    - uses: actions/checkout@v3
+      if: github.event_name == 'pull_request_target'
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - uses: actions/setup-go@v4
+      with:
+        go-version: '1.21'
+
+    - id: cache-ndk
+      uses: actions/cache@v3
+      with:
+        path: ${{ env.NDK_ZIP }}
+        key: ${{ env.NDK_ZIP }}
+    - name: download ndk
+      if: steps.cache-ndk.outputs.cache-hit != 'true'
+      run: wget https://dl.google.com/android/repository/${{ env.NDK_ZIP }}
+    - name: unzip ndk
+      run: unzip -d etc ${{ env.NDK_ZIP }}
+
+    - name: build
+      env:
+        BUILD_CHANNEL: ${{ inputs.release_type }}
+      run: make server-android
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: viam-server-android
+        path: ${{ env.TARGET }}
```

Comment:
I was about to ask how/where you were uploading for PR builds, then realized that's what this is probably for. Not a bad idea. May be worth doing for our other builds. Makes cleanup easier when it's an artifact associated with the run, rather than trying to track PR uploads in GCP perhaps.

Reply:
artifacts expire eventually but are neat for temporary use